src/app/pages/TokenDashboardPage/TokenTitleCard.tsx

```diff
@@ -49,7 +54,7 @@ export const TokenTitleCard: FC<{ scope: SearchScope; address: string }> = ({ sc
                 flexWrap: 'wrap',
               }}
             >
-              {token?.name ?? t('common.missing')}
+              {token?.name ? <HighlightedText text={token.name} pattern={searchTerm} /> : t('common.missing')}
```

Comment:
Indeed it's a bit of a pain. I'm not sure about the refactoring though, because there are use cases when there are multiple different highlighted terms active at the same time at the same page, coming from different sources..